onnxruntime/core/providers/tensorrt/tensorrt_compiler.cc

```diff
@@ -0,0 +1,462 @@
+    const std::string& name = trt_network->getOutput(i)->getName();
+    size_t bindingIndex = trt_engine->getBindingIndex(name.c_str());
+    nvinfer1::Dims dimensions = trt_engine->getBindingDimensions(static_cast<int>(bindingIndex));
+    bindingIndex -= num_inputs;
+    auto iter = output_map.find(name);
+    if (iter != output_map.end()) {
+      output_indexes[bindingIndex] = iter->second;
+    }
+    size_t dim_size = 1;
+    for (int j = 0, end = dimensions.nbDims; j < end; ++j) {
+      output_shapes[bindingIndex].push_back(dimensions.d[j]);
+      dim_size *= dimensions.d[j];
+    }
+
+    const auto& graph_output = model_proto.graph().output();
+    const auto& tensor_type = graph_output[i].type().tensor_type();
+    output_types[bindingIndex] = tensor_type.elem_type();
+
+    const auto& tensor_shape = tensor_type.shape();
+    if (tensor_shape.dim_size() == 1 && output_shapes[bindingIndex].back() == 1) {
+      output_shapes[bindingIndex].pop_back();
+    }
+  }
+
+  ORT_ENFORCE(trt_engine->getNbBindings() == (num_inputs + num_outputs));
+
+  // Save engine, context and input/output info to map
+  parsers_.emplace(fused_node->Name(), std::move(trt_parser));
+  engines_.emplace(fused_node->Name(), std::move(trt_engine));
+  contexts_.emplace(fused_node->Name(), std::move(trt_context));
+  input_info_[fused_node->Name()].push_back(input_indexes);
+  output_info_[fused_node->Name()].push_back(output_indexes);
+  output_info_[fused_node->Name()].push_back(output_types);
+  output_shapes_[fused_node->Name()] = output_shapes;
+
+  // Create function state
+  // TODO: remove default capture
+  node_compute_func.create_state_func = [=](ComputeContext* context, FunctionState* state) {
+    std::unique_ptr<TensorrtFuncState> p = std::make_unique<TensorrtFuncState>();
+    *p = {context->allocate_func, context->release_func, context->allocator_handle, parsers_[context->node_name].get(), engines_[context->node_name].get(), contexts_[context->node_name].get(),
+          input_info_[context->node_name], output_info_[context->node_name], output_shapes_[context->node_name], &tensorrt_mu_, stream};
+    *state = p.release();
+    return 0;
+  };
+
+  // Release function state
+  node_compute_func.release_state_func = [](FunctionState state) {
+    if (state)
+      delete static_cast<TensorrtFuncState*>(state);
+  };
+
+  // Create compute function
+  node_compute_func.compute_func = [](FunctionState state, const OrtCustomOpApi* api, OrtKernelContext* context) {
+    Ort::CustomOpApi ort{*api};
+    TensorrtFuncState* trt_state = reinterpret_cast<TensorrtFuncState*>(state);
+    const std::vector<int>& input_indexes = (trt_state->input_info)[0];
+    const std::vector<int>& output_indexes = (trt_state->output_info)[0];
+    const std::vector<int>& output_types = (trt_state->output_info)[1];
+    std::vector<std::vector<int64_t>> output_shapes = trt_state->output_shapes;
+    cudaStream_t stream = trt_state->stream;
```

Comment:
>stream [](start = 17, length = 6)

rename it to avoid dup with Compile(const onnxruntime::Node* fused_node, cudaStream_t stream,..)